rfcs/text/onChange.md

```diff
@@ -0,0 +1,128 @@
+* `SyntheticEvent.target` **MUST** have a value property, that property can be any value **UNLESS** the component supports
+multiple select, in that case the property must be an `Array` of values. If the component supports both single select
+and multiple select controlled by a prop the property **MUST** be an `Array`.
+* Components **MUST NOT** iterate over or replace props on the original `SyntheticEvent.target`
+* Components **MAY** reference values from the original `SyntheticEvent.target` or use `SyntheticEvent.nativeEvent.target` as they
+are synonymous.
+
+The approach suggested would modify the `SyntheticEvent` object that React's `EventPool` uses for performance optimisation.
+
+When validating this approach it was important to ensure that we didn't adversely affect the `EventPool`.
+
+* `SyntheticEvent` is constructed from a object in the `pool` [SyntheticEvent.js#L94](https://github.com/facebook/react/blob/ab1a4f249e61045d523ddbbfb840e868afbbf785/packages/react-dom/src/events/ChangeEventPlugin.js#L53)
+* `SyntheticEvent.target` is a pointer to `SyntheticEvent.nativeEvent.target` [SyntheticEvent.js#L94](
+https://github.com/facebook/react/blob/b1a03dfdc8e42d075422556553ffe59868150e95/packages/legacy-events/SyntheticEvent.js#L94)
+
+* React "nullifies" the event properties when released back to the pool [EventBatching.js#L34](https://github.com/facebook/react/blob/b1a03dfdc8e42d075422556553ffe59868150e95/packages/legacy-events/EventBatching.js#L34) => [SyntheticEvent.js#L179](https://github.com/facebook/react/blob/b1a03dfdc8e42d075422556553ffe59868150e95/packages/legacy-events/SyntheticEvent.js#L179)
+
+I could see no references to `.target` in the [legacy-events](https://github.com/facebook/react/search?q=target+path%3Apackages%2Flegacy-events&unscoped_q=target+path%3Apackages%2Flegacy-events) package.
+
+When validating this approach I reviewed the code of another popular component library, MaterialUI. [This is the same
+approach that they have adopted](https://github.com/mui-org/material-ui/blob/master/packages/material-ui/src/Select/SelectInput.js#L130).
+
+## Other considerations
+
+We currently have components that are controlled by a non-input field, for example `Date`. This is where a 3rd party date
+picker changes the underlying `value` of the `<input>`. There is an [outstanding bug in React](https://github.com/facebook/react/issues/13424) which means that `onChange` events attached to `<input>` won't trigger when changing the `value` with `setState`.
+
+For this reason, when programmatically setting `value` the event **MUST** be a plain object and not a `SyntheticEvent`.
+It *MUST* follow all the other requirements above.
+For example:
+```js
+const internalChange = value => {
+  const e = {
+    target: {
+      {
+      ...(name && { name }),
+      ...(id && { id }),
+      value
+    };
+    }
+  }
+  onChange(e);
+  };
+```
+
+# Drawbacks
+
+- It would be a breaking change and therefore would require a major version bump.
```

Comment:
I'm assuming the suggestion here is to make this change as part of v9?

Reply:
That's right. Tickets that are currently in progress have already started to adopt this process. We would need to have a cursory check of the other `_experimental_` components. If there are any changes required to `src/components` I propose we make those at a later date.

Reply:
That's right. Tickets that are currently in progress have already started to adopt this proposal. I think we should have a cursory check of everything in `_experimental_`. We can do this using the `fixture` screens that already exist. I suggest that any changes to components in `src/components` are made at a later date.